File: hangman_ruby/GuessLetter.rb
require 'Guess'

# A Guess that represents guessing a letter for the current Hangman game

class GuessLetter < Guess
  @guess

  def initialize(guess)
    @guess = guess
  end

  def makeGuess(game)
    game.guessLetter(@guess)
  end

  def to_s
    "GuessLetter[%s]" % @guess
  end
end

File: trunk/hangman_js/GuessLetter.js
"use strict"

var Guess = require("./Guess.js").Guess;

/**
 * A Guess that represents guessing a letter for the current Hangman game
 */
function GuessLetter(guess) {
  this.guess = guess;
}

GuessLetter.prototype = new Guess();
GuessLetter.prototype.constructor = GuessLetter;

GuessLetter.prototype.makeGuess = function(game) {
  game.guessLetter(this.guess);
}

GuessLetter.prototype.toString = function() {
  return "GuessLetter[" + this.guess + "]";
}

exports.GuessLetter = GuessLetter;
